main.go

```diff
@@ -245,17 +247,32 @@ func mainCore() error {
 			if err != nil {
 				return fmt.Errorf("RewindStakeDB failed: %v", err)
 			}
-			if stakedbHeight != lastBlockPG {
+			// stakedbHeight is always rewinded to height of zero even when lastBlockPG is -1
+			if stakedbHeight != lastBlockPG && stakedbHeight > 0 {
 				return fmt.Errorf("failed to rewind stakedb: got %d, expecting %d",
 					stakedbHeight, lastBlockPG)
 			}
 		}
 
+		block, err := dcrdClient.GetBlockHeader(blockHash)
+		if err != nil {
+			return fmt.Errorf("unable to fetch the block from the node: %v", err)
+		}
+
+		// If one block takes on average 5 minutes to mine, get the new blocks
+		// count that can be mined in the elaspedTime.
+		elapsedTime := time.Since(block.Timestamp).Minutes()
+		expectedHeight := int64(elapsedTime / 5)
+
+		if height > expectedHeight {
+			expectedHeight = height
+		}
+
 		// How far auxDB is behind the node
-		blocksBehind := height - lastBlockPG
+		blocksBehind = expectedHeight - lastBlockPG
 		if blocksBehind < 0 {
 			return fmt.Errorf("Node is still syncing. Node height = %d, "+
-				"DB height = %d", height, heightDB)
+				"DB height = %d", expectedHeight, heightDB)
```

Comment:
But we know the  height of the best block... Why guess it from the best block's timestamp?

Reply:
This part was introduced help to handle the case scenarios where dcrdata data migration is initiated when dcrd is not sync'd with the latest block in the chain. That difference in sync affects how we may handle data in bulk.
We don't have trustworthy way to fetch the latest block in the chain even in case scenarios when dcrd and dcrdata are almost simultaneously run together. Adding the `expectedHeight` value that is calculated from the best block timestamp means that we can roughly be able to estimate the latest block in the chain even when dcrd is not sync.

Reply:
```go
elapsedTime := time.Since(block.Timestamp).Minutes()
expectedHeight := int64(elapsedTime / 5)
```
Say this indicates the node's best block is 1 day old, `expectedHeight` as computed is the estimate of the *number of blocks dcrd has left to process to get up with the network*, not the networks estimated best block height.

The expected height would be `int64(elapsedTime / p.ChainParams.TargetTimePerBlock) + int64(block.Height)`.

Reply:
here the bigger value is used as what should be used to make decisions on how to handle the bulk data.

```go
if height > expectedHeight {
        expectedHeight = height
}
```

Reply:
But `expectedHeight` is not a height, it is an estimate of the remaining blocks that dcrd has to sync.  In plain words, what is it that `expectedHeight` should represent?

Reply:
`expectHeight` is simply an estimate of what block height value should be used to make  close to accurate decisions on how to handle bulk data.

The previous implementation was only accurate if dcrd was in sync with the latest block from the blockchain.

Reply:
Right, but it is not "an estimate of which height value", at least not as you are computing it now.

If `block` is dcrd's current best block that is somewhere behind the network's best block, and then you have:

```go
elapsedTime := time.Since(block.Timestamp).Minutes()
expectedHeight := int64(elapsedTime / 5)
```

Then `elapsedTime` is the time since dcrd's best block was mined.  So `elapsedTime / 5` is an estimate of the number of blocks dcrd is behind by, not the height of the network that dcrd is trying to reach.  Note in https://github.com/decred/dcrdata/pull/710#discussion_r221281566 I said I would expect you'd need to compute:

`int64(elapsedTime / p.ChainParams.TargetTimePerBlock) + int64(block.Height)`